crest/Assets/Crest/Crest/Shaders/Ocean.shader

```diff
@@ -308,15 +309,15 @@ Shader "Crest/Ocean"
 				// Sample shape textures - always lerp between 2 LOD scales, so sample two textures
 
 				// Calculate sample weights. params.z allows shape to be faded out (used on last lod to support pop-less scale transitions)
-				const float wt_smallerLod = (1. - lodAlpha) * _LD_Params[_LD_SliceIndex].z;
-				const float wt_biggerLod = (1. - wt_smallerLod) * _LD_Params[_LD_SliceIndex + 1].z;
+				const float wt_smallerLod = (1. - lodAlpha) * _CascadeData[_LD_SliceIndex]._weight;
```

Comment:
Will shaders optimise the fact we are sampling the same compute buffer over-and-over again into a single access?

Reply:
Thanks yeah i forgot about that and am hoping to have a stab at this this weekend. I'm not totally sure but i could imagine it is not optimal to do this. I could check the compiled code to say for sure but i think factoring it out is a good thing in general so that the functions dont rely implicitly on these driven values..

Reply:
I mostly did it. theres a variety of different cases (some sample with different array indices and etc). theres a few notable cases i left in ocean helpers. i think there might be more consolidation we need to do. also things may be a little different in the RPs in terms of how best to pass data around, so i'd propose settling it downstream before going further.